### ISSUE TITLE
Different RedisFailover's sentinels join together

### ISSUE THREAD

**Issue by dbackeus** [freshworks/redis-operator#6]
I'm reopening what was probably the most critical bug before the fork: https://github.com/spotahome/redis-operator/issues/550

It can result in complete data loss for a cluster. Not just in theory, it happened in our production k8s cluster. Eg. if cluster A gets mixed up with cluster B and cluster A ends up being elected master for all nodes, the previously existing data of cluster B will get completely overwritten by cluster A's.

I still believe the appropriate solution is to start relying on hostnames instead of injecting pod IP's into config as mentioned in https://github.com/spotahome/redis-operator/issues/550#issuecomment-2270582339

**Comment by samof76** [comment]
@dbackeus  Can you please simulate with those changes? in the #3

**Comment by dbackeus** [comment]
Bad timing, we ditched the operator in favour of our own templates two weeks back. If you're interested here's what they look like: https://github.com/reclaim-the-stack/get-started/commit/736a3af3e9acc9c48e92b6af2720775f568e6d56

So I won't be spending time on testing. This would be my feedback though:

Your PR should prevent sentinels from accidentally electing new masters across redis clusters, which is certainly better than what we have currently.

However, it can still be confusing that cross cluster sentinels start interacting, and it would not prevent the same problem from occurring on the Redis side.

Eg. `redis-cluster-A-1` is following `redis-cluster-A-2`. Some disaster occurs forcing some pods to reschedule and you end up with `redis-cluster-B-1` getting the IP of `redis-cluster-A-1` and causing `redis-cluster-A-2` to re-sync from cluster B.

The root problem is relying on IP's in the first place. Hostnames is really the way to go in Kubernetes since IP's are not reliable / deterministic.

**Comment by samof76** [comment]
I agree with you. This definitely not solve the entire problem, within the cluster, and across if you have unique names for your redis failover, which we have currently, as do not and will not run into this issue. I would definitely pick this one as feature to be enableb by a flag in the CRD. Do you have comments on this?

**Comment by dbackeus** [comment]
No additional comments, no.

**Comment by murtll** [comment]
I also faced this issue on bitnami/redis + AWS EKS.
Changing masterset for different sentinel clusters didn't help.
In my case it seems to be resolved with disabling [service links](https://github.com/bitnami/charts/blob/c72422d1e7270457250f37a10445be2e463d1f32/bitnami/redis/values.yaml#L1096-L1098) (requires bitnami/redis 17.17.0):

I don't know how exactly it was able to affect redis behavior, but I was thinking about any possibility for redis to discover other clusters. Service links got my eyes on them because they are are injecting env vars containing information about services in same namespace.

After disabling service links it's been a week without redis errors and sentinel tilts. Usually I had 1-2 falls per day.

Also as an additional step I've enabled [NetworkPolicies](https://github.com/bitnami/charts/blob/c72422d1e7270457250f37a10445be2e463d1f32/bitnami/redis/values.yaml#L1336-L1339) forbidding egress traffic from redis pod to all pods excep it's own cluster. It should prevent interactions between different clusters.

maybe it will be also helpful for people who struggle on https://github.com/bitnami/charts/issues/9689 and https://github.com/redis/redis/issues/10547
